src/libs/actions/Session/index.js

```diff
@@ -75,7 +85,7 @@ function signOutAndRedirectToSignIn() {
         Linking.getInitialURL().then((url) => {
             const reportID = ReportUtils.getReportIDFromLink(url);
             if (reportID) {
-                Report.setLastOpenedPublicRoom(reportID);
+                Onyx.merge(ONYXKEYS.LAST_OPENED_PUBLIC_ROOM_ID, reportID);
```

Comment:
Any reason we are doing this?

Reply:
Yes , according to [onyx documentation](https://github.com/Expensify/react-native-onyx#should-i-use-merge-or-set-or-both) **set** should be used only when deleting the key from store or resetting object or array of data. In our case we are only storing primitive value so to that case better is to use merge.